assignment-1/Functions.java

```diff
@@ -0,0 +1,271 @@
+            for (Players t : allPlayers) {
+                hash_Set.add(t.getTeam());
+            }
+            Integer srNo = 0;
+            for (String s : hash_Set) {
+                for (String t : hash_Set) {
+                    if (s != t) {
+                        srNo++;
+//                        System.out.println(s + " " + t);
+                        sb.append(srNo);
+                        sb.append(",");
+                        sb.append(s);
+                        sb.append(",");
+                        sb.append(t);
+                        sb.append(",");
+                        sb.append("null");
+                        sb.append("\n");
+                    }
+                }
+            }
+
+            pw.write(sb.toString());
+            pw.close();
+            System.out.println("fixture.csv file created!");
+
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+
+    }
+
+//    Third Task
+    public static void bowlersWithLeast40Wickets(String bowlersTeam) {
+        try {
+            for (Players p : allPlayers) {
+                if (p.getTeam().equals(bowlersTeam)) {
+                    if (p.getWickets() >= 40) {
+                        System.out.println(p.getName() + " " + p.getWickets());
+                    }
+                }
+            }
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    //Fourth Task
+    public static void searchPlayer(String playerName) {
+        try {
+            boolean flag = true;
+            for(Players p : allPlayers) {
+                if(p.getName().contains(playerName))
+                {
+                    flag = true;
+                    System.out.println("Name: " + p.getName());
+                    System.out.println("Team: " + p.getTeam());
+                    System.out.println("Role: " + p.getRole());
+                    System.out.println();
+                }
+                else {
+                    flag = false;
+                }
+            }
+            if(!flag) {
+                System.out.println("No Player found with name " + playerName);
+            }
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+//    Fifth Task
+
+    public static void runsAndWicketsScorer(String teamName) {
+
+        HashMap<String, Integer> runScorer = new HashMap<>();
+        HashMap<String, Integer> wicketTaker = new HashMap<>();
+        Map<String, Integer> sorted = new HashMap<>();
+        try {
+            for (Players player : allPlayers) {
+                if (player.getTeam().equals(teamName)) {
+                    runScorer.put(player.getName(), player.getRuns());
+                }
+                if (player.getTeam().equals(teamName)) {
+                    wicketTaker.put(player.getName(), player.getWickets());
+                }
+            }
```

Comment:
Since you already had a list of teams with their players stored, you could directly have used that extracted the top wicket-takers and run scorers from the team. this is a redundant task to generate the structure which already exists.